src/stack.ts

```diff
@@ -148,16 +148,19 @@ export class App
         const cli = AwsCdkCli.fromCloudAssemblyDirectoryProducer(this);
         this.appProps = props.args?.props;
         this.appOptions = props.args;
+        const lookupsEnabled = process.env.PULUMI_CDK_EXPERIMENTAL_LOOKUPS === 'true';
+        const lookups = lookupsEnabled && pulumi.runtime.isDryRun();
         try {
             // TODO: support lookups https://github.com/pulumi/pulumi-cdk/issues/184
-            await cli.synth({ quiet: true, lookups: false });
+            await cli.synth({ quiet: true, lookups });
```

Comment:
I just pushed a commit here that documents how this would work. Pretty much all apps will have pulumi resources due to the asset stuff.

https://github.com/pulumi/pulumi-cdk/pull/234